### PR TITLE
move release-schedule PR into proposed state

### PR DESCRIPTION
**Pull request by Flyingmana**
- [ ] Getting it moved into proposed state again by having 3 supporters

### CODE REVIEW

**Comment by Flyingmana** [comment]
/cc @OpenMage/lts-admins @sreichel @ADDISON74 @colinmollenhour 
did I miss any alternative?

**Comment by elidrissidev** [comment]
Not sure if it's in the scope of this rfc, but I think it's important to decide whether the releases will follow semantic versioning and mention that in the README, especially for people installing OM with composer since they probably expect that similar to the majority of other packages.

**Comment by Flyingmana** [comment]
All 3 alternatives are following semantic versioning, although alternative 1) is only indirectly mentioning it through the nginx reference.

**Comment by addison74** [comment]
I will do an analysis from the perspective of a neutral user, as otherwise I could become subjective. The 3 alternatives shown are a good starting point.

**Comment by fballiano** [comment]
I vote for alternative 3, although I wouldn't have a fixed monthly release, we may be in the middle of some project and the repo could be in a "not ready for release" state. Also, 12 releases per year seem to much, the releases would be really small and less "interesting" when we need to create buzz around it (although a small changelog is easier to manage for the store owners, 12 releases/year will anyway create some headaches).

I think 4 yearly releases it would be great.

**Comment by Flyingmana** [comment]
> I vote for alternative 3, although I wouldn't have a fixed monthly release, we may be in the middle of some project and the repo could be in a "not ready for release" state. Also, 12 releases per year seem to much, the releases would be really small and less "interesting" when we need to create buzz around it (although a small changelog is easier to manage for the store owners, 12 releases/year will anyway create some headaches).
> 
> I think 4 yearly releases it would be great.

I take this as an "approved for moving into the next step, where we then can vote on it"?

**Comment by fballiano** [comment]
@Flyingmana if you want we could have a 3.1 (release monthly) and 3.2 (release quarterly). There would be a 4 (main + only backport extremely important stuff to the "previous" version) but that's kinda covered by #6 so I don't know about that.

**Comment by addison74** [comment]
Maybe we should approach from another perspective. For example, at a PRs number or at a number of months, obviously excluding the situation where a new emergency version is desperate needed. If we have 100 PRs in a month we can release a new version every month, but if we have 10 PR, we release it every 4 months. I take in consideration the effort of those who make this process possible.

**It would be good to discuss.**
Would it be possible to have two repositories in OpenMage, one for OpenMage and another for Magento LTS? Thus we would separate these projects once and for all, each one will have its own course. How it is now seems mixed to me, in addition, it's also tiring to keep hearing that a PR goes sometimes there, sometimes beyond, that we can do a BC and many others. OpenMage becomes a main project and will provide Magento LTS with certain PRs if needed.

**Comment by fballiano** [comment]
> Maybe we should approach from another perspective. For example, at a PRs number or at a number of months, obviously excluding the situation where a new emergency version is desperate needed. If we have 100 PRs in a month we can release a new version every month, but if we have 10 PR, we release it every 4 months. I take in consideration the effort of those who make this process possible.

mmmm interesting, it makes a lot of sense

> **It would be good to discuss.** Would it be possible to have two repositories in OpenMage, one for OpenMage and another for Magento LTS? Thus we would separate these projects once and for all, each one will have its own course. How it is now seems mixed to me, in addition, it's also tiring to keep hearing that a PR goes sometimes there, sometimes beyond, that we can do a BC and many others. OpenMage becomes a main project and will provide Magento LTS with certain PRs if needed.

that's a whole different Pandora's vase :-) it's an interesting discussion to have but I think it's outside the scope of this specific RFC

**Comment by elidrissidev** [comment]
The 3 proposed solutions are good IMO. So this is ready to move to the next step.

**Comment by Flyingmana** [comment]
I do count 3 supporters now, this means we are ready to move this to proposed and start the actual voting phase, right?

**Comment by sreichel** [comment]
> If we have 100 PRs in a month we can release a new version every month, but if we have 10 PR, we release it every 4 months.

Just picked this comment ... and ... https://github.com/OpenMage/organizational-internal/issues/14

I can understand that you want to follow semantic versioning, but can we do this?

If something went wrong can we provide a "hot-fix" release in time? Who decides what requires a patch/minor/major release?

For me it still looks like some _overhead_.

Suggestion: Adding monthly releases ... or earlier depending on count of commits 

PRO: less conflicts/trouble to get PRsl merged
CONS: no semantic versioning

**Comment by Flyingmana** [comment]
> Who decides what requires a patch/minor/major release?

that would be something for a different RFC. Till then I would say, everything which is a breaking change for any user/Theme/ExtensionVendor, it has to go into a new Major Release. A single theoretical Case should be enough for it.

Iam undecided about minor versions, as long as we regularly merge in new features, it would make sense to release them as minor versions, which means usually only the newest branch gets minor releases, everything else only bugfix releases.

But yes, that should go into a different RFC

**Comment by elidrissidev** [comment]
> Who decides what requires a patch/minor/major release?

Think the [semver site](https://semver.org/) already defines that.

**Comment by fballiano** [comment]
> If something went wrong can we provide a "hot-fix" release in time? Who decides what requires a patch/minor/major release?
> 
> For me it still looks like some _overhead_.

true, but at the moment we didn't have any trouble deciding "ok let's release", and also the "which release number are we bumping too" went smoothly, so I wouldn't worry too much about that :-)

**Comment by sreichel** [comment]
> Think the [semver site](https://semver.org/) already defines that.

I know. But do or can we follow this. (just a question)

**Comment by elidrissidev** [comment]
Can this move forward now?

**Comment by colinmollenhour** [comment]
I updated Alternative 1, I hope it's ok, it was based on a suggestion I had proposed at one point and I don't think it was very popular anymore and wasn't very well defined anyway.. The new alternative is my best proposal based on observations from the past three years.

File: proposed/0002-release-schedule.md
Comment (fballiano) [inline proposed/0002-release-schedule.md]: these sections should be removed, they're not filled

File: proposed/0002-release-schedule.md
Comment (fballiano) [inline proposed/0002-release-schedule.md]: I think these cross with https://github.com/OpenMage/rfcs/pull/6, it could be ok (I like this alternative the way it's written now) but these "where do patches go and for how long" sections are missing from alternative 3, which kinda doesn't make sense anymore.

File: proposed/0002-release-schedule.md
Comment (colinmollenhour) [reply proposed/0002-release-schedule.md]: Updated although Alternative 1 and 2 leave some unresolved questions..

File: proposed/0002-release-schedule.md
Comment (colinmollenhour) [reply proposed/0002-release-schedule.md]: Ahh, thanks, I forgot about that PR, these are indeed very similar.. I think branching and release strategy are so closely related they should both be resolved together, no?

File: proposed/0002-release-schedule.md
Comment (addison74) [inline proposed/0002-release-schedule.md]: If you display the rich page the image is missing. The link is not working.

Comment (colinmollenhour) [reply proposed/0002-release-schedule.md]: Fixed.

File: proposed/0002-release-schedule.md
Comment (colinmollenhour) [reply proposed/0002-release-schedule.md]: Yes, Alternative 3 was missing a lot of detail and is not necessarily different from Alternative 1 except the more rigid timing of releases so I removed it.
@Flyingmana or anyone else feel free to add it back, but with more detail about maintenance of old versions. :)

Comment (fballiano) [reply proposed/0002-release-schedule.md]: I think (actually, if I'm not mistaken, that's also @Flyingmana opinion) that this PR should be only about release schedule (which also defines how many branches we are maintaining) and then the other one would be different and still needed.

but the way alternative 1 is phrased now, makes alternative 3 useless

Comment (fballiano) [reply proposed/0002-release-schedule.md]: also, this line

`- Old major releases get their own branch only for the purpose of back-porting PATCH fixes (v19, v20, v21, etc)  `

should have a precise number of branches that patches are ported, otherwise it's again open to intepretation, which is what we're trying to avoid.

Comment (fballiano) [reply proposed/0002-release-schedule.md]: these modifications are not big but change the RFC quite a bit and I think they should have been done in the drafting phase, we should not vote now until we get everything fixed for everybody.

**Comment by addison74** [comment]
Now that you have simplified things and eliminated alternative 3, I vote for alternative 1.

Realistically speaking, do you think we will ever have a major release? What could this contain to name it major? So far, apart from small features, we have mainly patched Magento to bring it to 2023. 

I could consider the replacement of the unmaintained Zend library with ZF1-Future a major change. Another could be to specify that OM is tested and fully functional with PHP 8.2.

File: proposed/0002-release-schedule.md
Comment (Flyingmana) [reply proposed/0002-release-schedule.md]: > these modifications are not big but change the RFC quite a bit and I think they should have been done in the drafting phase, we should not vote now until we get everything fixed for everybody.

as long as this PR is not merged, it is not yet in proposed state, therefore its still in drafting phase.

**Comment by Flyingmana** [comment]
a reminder, because people are already wanting to vote.
We are in the phase before voting, where the Draft is finished and moved into the proposed state.
The voting starts afterwards.

**Comment by colinmollenhour** [comment]
Attempt to define version increments which differ slightly from semver.org for our purposes:

- MAJOR version when you make incompatible API changes
  - public methods with changed signatures (e.g. new required parameter)
  - removed code that may still be used by some (e.g. unused lib/Zend code or IE8 compat)
  - rename things that could affect users (e.g. xml namespace)
  - dependencies updated in major ways (e.g. dropping support of PHP 7.4 or MySQL 5.6)
- MINOR version when you add functionality in a backwards compatible manner
  - changed code that is very unlikely to have any effect on large majority of users although not guaranteed
  - typical usage is not expected to be affected, only if user was not following best practices
  - no regressions foreseen, but there is always a risk a new feature affects some in unexpected ways
  - performance/stability is improved significantly with little or no negative impacts expected
  - dependencies updated to fix bugs or bring in minor improvements
- PATCH version when you make backwards compatible bug fixes
  - best case: fix has zero negative impart
  - worst case: security fix is urgent enough to justify potentially breaking user's stores in a small way (e.g. improving malicious string filtering)

Thoughts?

As written above, probably more things should have been counted as major in the past.. Maybe we should have a "next" branch for major features so that "main" can still be the default and we can limit major releases to 1-2 per year?

**Comment by colinmollenhour** [comment]
I pushed some more updates for Alternative 1 based on some more contemplation about release cycles and branching. [(formatted)](https://github.com/OpenMage/rfcs/blob/cf9fddcc5a87d01b259c1126214ffa6e731b387e/proposed/0002-release-schedule.md)

There are a lot of details missing from Alternative 2 - if it has any strong proponents I think it needs to be fleshed out more - if not I propose we remove it and refine Alternative 1 for a Yes/No vote (although a No result would leave us back at square one).

**Comment by fballiano** [comment]
So, with this new version, if I understood correctly, 1.9.4.x would become v19, 20.0 would become main, there would be no next branch at the moment.

Although probably v19 should be derived from 19.4.21 and not using the "composer" thingy? because in this moment the situation is right in the middle...

**Comment by justinbeaty** [comment]
I like Alternative 1, and I think it accomplishes everything discussed in https://github.com/OpenMage/magento-lts/discussions/2327. I think a simple yes/no vote would be a good idea. 

From my understanding, this is what we would do:

1. Rename 20.0 branch -> main, set as default branch

2. Immediately follow steps from "Just before a MAJOR release is tagged":
    - Create branch from main called "20.0"
    - Create branch from main called "next"

3. Then, PRs will need their target branch changed. All PRs on 1.9.4.x would retarget to main. PRs on 20.0 will either target main (already the case because we would rename 20.0 -> main), or next depending on if they're breaking.

Pros:
- We have an always up to date branch for the latest release.
- We also have the experimental branch "next" that has major breaking changes.
- These two points allow someone to use the latest code via composer without worrying about breaking changes being introduced.
- Most importantly, it lets 1.9.4.x go into patch only mode and out of support in 2 years, which lines up pretty well with the initially promised 5-year support period.

Some minor points with branch naming and our current versions:
- We would now be on v21, since we would have both a 20.0 branch and a main branch.
    - Perhaps we could take this time to merge any major breaking changes like https://github.com/OpenMage/magento-lts/pull/2730.
- We already have branches named 1.9.4.x and 20.0, which we decided to not rename. So when v22 is released, I would say to branch main to "21.x". This keeps everything consistent except for v20, but since that never had a minor release, "20.0" is okay.

**Comment by fballiano** [comment]
mmm the thing is I wouldn't want to have more than one active branch + a patch branch (the v19 of today), if we start having a next, v21, v20, v19, main, that's a no no.

if there's a possibility of 2 active branches.

the new version of this RFC is very complete but also kinda complex and I don't know if it cover what I'd like from it, and what's actually doable with our manpower.

File: proposed/0002-release-schedule.md
Comment (fballiano) [inline proposed/0002-release-schedule.md]: these lines should be removed

File: proposed/0002-release-schedule.md
Comment (fballiano) [inline proposed/0002-release-schedule.md:47]: if there are 2 major releases per years and we have to support each branch for 2 years it mean we'll have 4 active branches + main and next?

File: proposed/0002-release-schedule.md
Comment (fballiano) [inline proposed/0002-release-schedule.md]: this should already be in the RFC, otherwise it's not superseeding https://github.com/OpenMage/rfcs/pull/6 completely

**Comment by justinbeaty** [comment]
@fballiano So as I understand the RFC as it reads now, we might have this:

- `1.9.4.x` - Patch only. LTS ends January 2025.
- `20.0` - Patch only. LTS ends January 2025.
- `main` - The main active branch, everything that's not breaking. For the `21.x` releases.
- `next` - Fewer PRs, only breaking changes. Commits here may not end up in a release for up to a year.

Perhaps it is too much? It does seem to make sense to me, though. Edit: So in summary, there would be two active branches plus several patch only branches.

**Comment by fballiano** [comment]
the goal for https://github.com/OpenMage/rfcs/pull/6 was to bring work down to one active branch + one patch, I don't know, to me this is creating more confusion.

**Comment by Flyingmana** [comment]
> * `20.0` - Patch only. LTS ends January 2025.

not sure what the RFCs say in the end, but 20.0 was not an LTS release and intended to end support half a year after the next release

**Comment by Flyingmana** [comment]
With the back and forth of support times, maybe it is better to keep multiple options with different alternatives and vote between them instead of reducing to a single option.

**Comment by Flyingmana** [comment]
The big difference between Alternative 1 and 3 was:
* 1 ) has 2 supported branches and Versions which are started in parallel. The main one, which has all new developments, and gets new features added to the one major release continuosly
* 3 ) only did support for a single Branch / Major version, which implies that as soon as there is a new Major Release, the previous one stops with Support.

Sounds similar at first, but is very different. We had for both approaches people who requested constellations which match them.
Interestingly Alternative 1 ) is technically not fully correct regarding SemVer

**Comment by justinbeaty** [comment]
> not sure what the RFCs say in the end, but 20.0 was not an LTS release and intended to end support half a year after the next release

Per Colin's schedule, every [major] release would be supported for two years [after the last minor version]. Taking into the idea of Ubuntu's release schedule, perhaps only every other release gets patches. I.e. v19, v20, v22, etc (Note I used even ~~years~~ versions, but also included v19 since that should remain LTS for a while yet). That also alleviates some concerns by @fballiano with too many active patch only branches. Ultimately I would give weight to the maintainers' opinion on this.

**Comment by addison74** [comment]
I would not complicate things. Two branches, one LTS (currently 19) and another MAIN (currently 20) are more than enough. As LTS wants not to create BC problems, I would make it subsidiary to MAIN. When a PR is created, an evaluation is made by maintainers if it goes only in MAIN, LTS or both. As it has been done until now, what goes in one is cherry-picked in the other. So far we were only discussing the placement of PR in branches.

Regarding the releases of new versions, I would come up with the proposal to have two archives openmage/openmage-lts or om/om-lts/omlts. Regarding major or minor releases, we can use numbers to make the difference between major and minor. I think that openmage-2.13 could identify the MAIN branch, with its second major and the 13th minor modifications. Maybe it is not the most inspired, but it must be as simple as possible and it must be intuitive. It is obvious for all that there will be more minors than majors. using the year is not a solution. Although we could also use version like this omlts-23.1 which would show the first release in 2023. In this case we don't know anything about the major nature of the version from the name.

As far as I am concerned, I was never interested in the LTS version and I only referred to the MAIN version, but it must be maintained to be able to be used with old, non-updated extensions. I don't know how much longer some stores can withstand the new PHP versions, but this is the problem of others. In this way we better focus our efforts on what OpenMage should be at this moment, but without neglecting its past.

**Comment by justinbeaty** [comment]
Building on my previous comment; to solve the patch only branches, what if the RFC was like this:

- Alternative 1a
    - PATCH fixes will only be back-ported to MAJOR versions that are less than 2 years (730 days) old
- Alternative 1b
    - PATCH fixes will only be back-ported to *every other* MAJOR version. I.e. v19, v20, v22, v24, etc that are less than 2 years (730 days) old
- Alternative 1c
    - PATCH fixes will only be back-ported to *every fourth* MAJOR version. I.e. v19, v20, v24, v28, etc that are less than 2 years (730 days) old
- Alternative 1d
    - PATCH fixes will only be back-ported to the most recent [non main branch] MAJOR release, after the previous LTS release is EOL. So, whatever the most recent v* branch is when v19 expires, will become the new LTS release.

I think that gives some option to vote on, no? 1b could omit v20, and be v19, v21, v23, etc, or we skip one to get to even numbers so v19, v22, v24, etc. Similar with 1c.

Regarding the next branch, I still think that's a good idea, because it lets us have an unstable branch where we can merge things that need testing, such as release builders, and other major breaking changes.

Edit: clarified some of the options above.

**Comment by fballiano** [comment]
I don't understand the porting patches to every X versions.

But I think we shouldn't have a v21 until v19 is EOL, that would avoid any question.

**Comment by justinbeaty** [comment]
> I don't understand the porting patches to every X versions.

I proposed it as an option to reduce the amount of active patch only branches. It would simply mean that v20 would get patches after development ends, while v21 would not. So if a user wants to upgrade less often, they should only upgrade to even numbered versions.

> But I think we shouldn't have a v21 until v19 is EOL, that would avoid any question.

The problem becomes that we should not make any breaking change until v19 is EOL (2024-2025?). Otherwise we continue to not adhere to semver.

**Comment by fballiano** [comment]
v19 would become patch-only without any backport

**Comment by justinbeaty** [comment]
> v19 would become patch-only without any backport

Indeed, I've always agreed with this. I also agree that we should have a main branch. So I think we mostly agree with everything already. I guess the main questions are:

1. How often do we make major releases?
2. For what previous major releases do we apply patches?

I think the option 1d is probably most similar to what you'd like. In this case, we adhere better to semver with major versions upon breaking changes, but we only ever have one patch only branch.

**Comment by colinmollenhour** [comment]
> How often do we make major releases?

For Alternative 1 (A1) - as-needed with a maximum of 2 per year. Personally I don't want to commit to a fixed schedule and then end up making releases just to abide by the schedule, or delaying a major feature because it barely missed a specific date.

> For what previous major releases do we apply patches?

For A1 - I believe it makes sense that as the schedule is flexible, the backporting should be based on age and not number of releases. I don't think we should complicate it with skipping major releases since the whole point of backporting is to give users a known amount of time to be "safe" after they upgrade. Inevitably some users will upgrade to a release that gets skipped and then be in trouble when a patch is released because they are on an in-between version.

Note that significant security vulnerabilities (PATCH) are expected to be very infrequent - the definition of PATCH being much more strict than what we've been putting into 1.9.4.x to date. Probably a good threshold would be that if it isn't fixing a regression from a MINOR update or it doesn't warrant a CVE, it doesn't get a PATCH.

> mmm the thing is I wouldn't want to have more than one active branch + a patch branch (the v19 of today), if we start having a next, v21, v20, v19, main, that's a no no.

Please add an Alternative 3 with your proposal. 😁 A branch is just a pointer to a commit. Yes, we should keep the list short but they serve a purpose and we only would be committing to keeping "main" and "next", the other branches can be deleted once they are EOL.

File: proposed/0002-release-schedule.md
Comment (colinmollenhour) [reply proposed/0002-release-schedule.md]: Done.

File: proposed/0002-release-schedule.md
Comment (colinmollenhour) [reply proposed/0002-release-schedule.md:47]: Yes, that is correct, which is why it is very important to be very strict with what is considered a PATCH. I updated the definition accordingly.

File: proposed/0002-release-schedule.md
Comment (colinmollenhour) [reply proposed/0002-release-schedule.md]: I updated for clarity

**Comment by colinmollenhour** [comment]
Great feedback guys! I pushed a minor update to the RFC based on your feedback.

If anyone wants to keep Alternative 2 or add a new Alternative 3, please indicate as such and let us know how much time you need to update the RFC, otherwise I propose we remove Alternative 2 and make this a Yes/No vote if there are no more changes added in the next 3 days.

**Review by colinmollenhour — APPROVED** [review]
I removed alternative 2 as there was no request to keep it and cleaned up the rest and in my opinion this is ready to move to a vote so it has my support.

File: proposed/0002-release-schedule.md
Comment (elidrissidev) [inline proposed/0002-release-schedule.md]: I think these are temporary branches created by @fballiano for the next release.

Comment (fballiano) [reply proposed/0002-release-schedule.md]: they will be removed in a couple of days max

File: proposed/0002-release-schedule.md
Comment (addison74) [inline proposed/0002-release-schedule.md:83]: I think we should limit ourselves to two subsets of numbers used in a version, for example 20.1, considering that we have 6 - 12 minor versions per year. The 3-set of number is no longer valid.

Comment (justinbeaty) [reply proposed/0002-release-schedule.md:83]: We have previously not been using SemVer, but this RFC indicated that we should start. So MAJOR.MINOR.PATCH is needed.

**Review by justinbeaty — APPROVED** [review]

**Comment by justinbeaty** [comment]
I think this is a great RFC, it will have my vote.

File: proposed/0002-release-schedule.md
Comment (addison74) [reply proposed/0002-release-schedule.md:83]: In this case I agree this format.

**Review by addison74 — APPROVED** [review]

**Comment by addison74** [comment]
It must be approved by "green people" only.

File: proposed/0002-release-schedule.md
Comment (fballiano) [reply proposed/0002-release-schedule.md]:
```suggestion
```

those branches do not exist anymore

**Comment by colinmollenhour** [comment]
@Flyingmana I'd like your blessing on this as it was your PR to begin with and I sorta hijacked it.. :)  If you have no objections or additional changes to make please merge it and create the PR for "accepted" so everyone can vote. Looking forward to having this settled!

**Comment by Flyingmana** [comment]
it has my full blessing, for me it was just important to have a democratic base behind the decision, which we have now.

But have currently 9 private things I need to finish before I can check back on it and do something

**Comment by fballiano** [comment]
I've re-read everything and I like it, just a question @colinmollenhour, can we avoid having a "next" branch if we don't have any major development going on, and create it when needed?

**Comment by colinmollenhour** [comment]
> ... can we avoid having a "next" branch if we don't have any major development going on, and create it when needed?

I think that would make it difficult for non-admins to create MAJOR update PRs based on the next branch and therefor discourage proper use of "next" as the merge base for the PR. I suppose you're concerned about "next" getting significantly behind "main" when there is not a lot of active development on MAJOR updates?

**Comment by fballiano** [comment]
> I think that would make it difficult for non-admins to create MAJOR update PRs based on the next branch and therefor discourage proper use of "next" as the merge base for the PR.

true, although I guess people would open the PR on the main branch, we would say "great, this is a major change, let's create the next branch and rebase the PR", it would seem like the normal flow, what do you think?

> I suppose you're concerned about "next" getting significantly behind "main" when there is not a lot of active development on MAJOR updates?

no it's actually more about having to update it constantly and maybe we don't even need it for many months

**Comment by colinmollenhour** [comment]
An explicit rebase step (if that's what you meant) might not even be necessary, I think just changing the PR is all that is needed.

However this clause would be broken if there was not always a "next" branch:

>  Bleeding-edge users and can easily use "dev-main" or "dev-next" via composer indefinitely.

Perhaps a Github action could auto-merge all "main" updates into "next" if and only if it is possible with a fast-forward? Then the manual maintenance would only start when next is different from main.

**Comment by colinmollenhour** [comment]
It looks like there are a few possibly relevant options already:

- Triggered by user comment: https://github.com/marketplace/actions/fast-forward-pr
- Triggered by cron: https://github.com/marketplace/actions/merge-fast-forward-only

So maybe this could just be considered later as an incremental improvement on the workflow after it is fully realized..

**Comment by fballiano** [comment]
mmmm ok, I still would prefer to have the next branch when needed but these action would work perfectly since there would be no conflicts ;-)

**Review by fballiano — APPROVED** [review]

**Comment by fballiano** [comment]
we've 2 green checks and 2 grey ones, can we try to move this forward?

**Review by mark-netalico — APPROVED** [review]

**Review by elidrissidev — APPROVED** [review]

**Comment by fballiano** [comment]
Time to merge and vote! how do we vote anyway? 😅

**Comment by elidrissidev** [comment]
From the repository README:
> start a PR moving it into the accepted state

> the PR has to stay open for at least 60 Days
> minimum 10 Votes
> having at least 70% accepting votes

**Comment by fballiano** [comment]
but the rule for merging was 3 green checks, I didn't force the merge, the button was green, and https://github.com/OpenMage/rfcs/pull/4 didn't have 10 votes, that's why I was pretty sure it was safe to merge...

**Comment by elidrissidev** [comment]
> but the rule for merging was 3 green checks, I didn't force the merge, the button was green, and #4 didn't have 10 votes, that's why I was pretty sure it was safe to merge...

This PR is fine, my comment was about the voting phase. I suppose create a PR that moves the file from "proposed" folder to "accepted", and that's where the final voting can be done.

**Comment by fballiano** [comment]
ah yes sure! 😅 sometimes I freak out I did some weird mistake

**Comment by elidrissidev** [comment]
Actually I'm a bit confused too, It seems that the last RFC went through one step only.

**Comment by colinmollenhour** [comment]
I started a new PR to move to accepted. Please vote there: https://github.com/OpenMage/rfcs/pull/10

**Pull request merged**
